scripts/entity/components/attack_components/base_attack_component.gd

```diff
@@ -1,11 +1,46 @@
 class_name BaseAttackComponent extends Node
 
+signal on_target_change(previous_target, current_target)
+
+## The Area2D that the character locks onto the closest target.
+@export var lock_on_area: Area2D
+
+## The node to compare against the targets.
+@export var measurement_position: Node2D
+
+
+## The current target that is closest in the area.
+var _current_target: Node2D = null
 
 # Called when the node enters the scene tree for the first time.
 func _ready() -> void:
-	pass # Replace with function body.
+	on_target_change.connect(target_changed)
 
 
 # Called every frame. 'delta' is the elapsed time since the previous frame.
-func _process(_delta: float) -> void:
-	pass
+func _physics_process(_delta: float) -> void:
+	update_closest_target()
```

Comment:
Doing this every frame feels like it'll be really rough. Is it possible for us to use anything specific to Area2D to do this? Maybe only check when something new enters? Or if what we were targeting leaves the range?

Reply:
I agree with this sentiment. I dont know if there is a good way to check though. Since if you exit / enter it doesnt let you know where things moved within what you have.

Reply:
We could potentially run a coroutine that updates every half-second or so? It wouldn't be 100% the most accurate, but should be accurate enough. Have it run a check if there is a closer/new target, and then if not we stop running the coroutine.

Reply:
Yea. My opinion was either do something like a coroutine or just event driven such as input. created a variable length coroutine

Reply:
I think coroutine might work better due to having a need for a reticle to be brought on, and it might be weird to show the last targeted with reticle even out of range. Granted maybe that's just something that is reticle-specific fix.

Reply:
From my understanding await was godot's coroutine